core/templates/services/insert-script.service.ts

```diff
@@ -22,41 +22,74 @@ import {downgradeInjectable} from '@angular/upgrade/static';
 export enum KNOWN_SCRIPTS {
   DONORBOX = 'DONORBOX',
   UNKNOWN = 'UNKNOWN',
+  MATHJAX = 'MATHJAX',
 }
 
 @Injectable({
   providedIn: 'root',
 })
 export class InsertScriptService {
```

Comment:
```
// The loading method continues only if the script is not in partiallyLoadedScripts.
// This is to prevent the same script from creating multiple promises to load. This can
// happen for both the MATHJAX and DONORBOX scripts.
// The same script can create multiple promises as we call loadScript in the ngOnInit
// function of the component needing the script. If on exploration editor, you open the
// math expression editor, the MATHJAX script starts loading. If you close the editor
// before it loads completely and reopen it again, another promise will be created
// for the MATHJAX script due to loadScript being called by the ngOnInit function again.
// Hence, we need the partiallyLoadedScripts Map and these checks to prevent this from
// happening.
``` 

What do you think of this comment to explain the double loading logic? Do you think this needs to be shortened?

Reply:
No, this is great. Long comments are fine. The main thing is that the comment provides useful context/info for the developer, which this does.

So, LGTM. Thanks! :)